sky/engine/core/script/dart_service_isolate.cc

```diff
@@ -55,49 +50,10 @@ const uint8_t* GetSymbol(Dart_NativeFunction native_function) {
   return g_natives->GetSymbol(native_function);
 }
 
-}  // namespace
-
-class Resources {
- public:
-  static const int kNoSuchInstance = -1;
-  static int ResourceLookup(const char* path, const char** resource) {
-    ResourcesEntry* table = ResourcesTable();
-    for (int i = 0; table[i].path_ != NULL; i++) {
-      const ResourcesEntry& entry = table[i];
-      if (strcmp(path, entry.path_) == 0) {
-        *resource = entry.resource_;
-        DCHECK(entry.length_ > 0);
-        return entry.length_;
-      }
-    }
-    return kNoSuchInstance;
-  }
-
-  static const char* Path(int idx) {
-    DCHECK(idx >= 0);
-    ResourcesEntry* entry = At(idx);
-    if (entry == NULL) {
-      return NULL;
-    }
-    DCHECK(entry->path_ != NULL);
-    return entry->path_;
-  }
+EmbedderResources resources(
```

Comment:
Does this create a static intializer and an exit-time destructor?  If so, we might want to use a singleton pattern that avoids those.